src/arweave/composite-client.ts

```diff
@@ -1253,6 +1258,88 @@ export class ArweaveCompositeClient
       Promise.all(secondaryResults);
     }
 
+    if (config.ARWEAVE_PEER_CHUNK_POST_MIN_SUCCESS_COUNT_STRING > 0) {
+      const peerUrls = this.selectPeers(
+        config.ARWEAVE_PEER_CHUNK_POST_CONCURRENCY_LIMIT_STRING,
+        'weightedPostChunkPeers',
+      );
+
+      const peerPostLimit = pLimit(
+        config.ARWEAVE_PEER_CHUNK_POST_CONCURRENCY_LIMIT_STRING,
+      );
+      let peerSuccessCount = 0;
+
+      await Promise.all(
+        peerUrls.map((peerUrl) =>
+          peerPostLimit(async () => {
+            if (peerSuccessCount >= chunkPostMinSuccessCount) return;
+            try {
+              const resp = await axios({
+                method: 'POST',
+                url: `${peerUrl}/chunk`,
+                data: chunk,
+                timeout: DEFAULT_CHUNK_POST_RESPONSE_TIMEOUT_MS,
+                signal: AbortSignal.timeout(
+                  DEFAULT_CHUNK_POST_ABORT_TIMEOUT_MS,
+                ),
+                headers: originAndHopsHeaders,
+                validateStatus: (status) => status === 200,
+              });
+
+              if (resp.status === 200) {
```

Comment:
Is this redundant with `validateStatus` above?

Reply:
good catch, yes for sure it shouldn't be needed twice

Reply:
fixed